historical.js

```diff
@@ -102,6 +102,48 @@ module.exports = function (schema, options) {
         historical.save(next);
     });
 
+
+    schema.pre('findOneAndUpdate', function (next) {
+        var query = this.getQuery();
+        this.model.findOne(query).exec().then(function(doc) {
```

Comment:
It looks to me like we never interact with the updates being made, instead this saves a historical record based on the document as it is in the database before the update.

Reply:
You're right, I was thinking this was the desired functionality but I now realize this wouldn't make sense given the way the pre save hook works currently since the document in the db should already have a historical record made upon creation. Essentially we want the most recent historical entry to match the current database entry right?

Reply:
Exactly. Historical records contain the most recent changes made to a document.